### ISSUE TITLE
INTERSVR hangs immediatly

### ISSUE THREAD

**Issue by egrath** [joncampbell123/dosbox-x#4393]
### Describe the bug

Microsoft Interlink server (INTERSVR.EXE) from MS-DOS 6.22 hangs immediatly upon running it. The debug log shows consecutive lines containing:
```
ERROR CPU:Illegal Unhandled Interrupt Called 6
```
Advices given in other bug reports like running ```LOADFIX``` first made no difference.

My general intend is to run INTERSVR.EXE in DOSBOX-X and provide access to the host filesystem to a serially attached physical computer running MS-DOS.

### Steps to reproduce the behaviour

1. Use the latest daily build
2. Run INTERSVR.EXE from MS-DOS 6.22

### Expected behavior

INTERSVR is started successfully and serves mounted drives via serial link

### What operating system(s) this bug have occurred on?

Windows 11

### What version(s) of DOSBox-X have this bug?

2023.05.01 and also latest daily builds

### Used configuration

```ini
[sdl]
windowresolution=1920x1440

[dosbox]
machine=svga_et4000
memsize=8
vmemsize=1

[render]
aspect=true
glshader=sharp

[serial]
serial1=direct realport:COM1

[autoexec]
mount C: DOS_C
mount D: SHARED
C:
cd INTERLNK
INTERSVR /COM1 /B
```

### Output log

```text
ERROR CPU:Illegal Unhandled Interrupt Called 6
```

### Have you checked that no similar bug report(s) exist?

- [X] I have searched and didn't find any similar bug report.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by dbjh** [comment]
You may not have the right hardware, but have you considered connecting the computers with a LapLink cable? Using a parallel connection is significantly faster than a serial one. Additional hardware is a PCIe card, preferably IEEE 1284 compliant.

**Comment by egrath** [comment]
Yes, already thought about this and would be possible on the hardware side of things. But even with the serial port disabled and parallel port enabled on the DOSBOX side, INTERSVR hangs immediatly upon start (Microsoft Interlink supports serial and parallel connections, though).
